Automattic/sensei
# Use global styles colors in Learning Mode

Fixes [#5544](https://github.com/Automattic/sensei/issues/5544), #5543

### Changes proposed in this Pull Request

* Use 'Link color' as the primary color
* Generate CSS variables for colors set via global styles, and use them as the base text/background/primary color
* Generate CSS variables for block wrappers in the editor and for the frontend when link or background color is set for that block

The DOM then will look something like this: 
```html
<body style="--primary-color: blue"> // Default Link color is Blue
    <sensei-button /> // Blue
    <group style="--primary-color: red">  // Set the Link color to Red
        <sensei-button /> // Red
    </group>
</body>
```

### Testing instructions

* Use a block theme, edit lesson template in the site editor
* Open the 'Styles' sidebar and change Background, Text, Link colors
* Primary colors in the template (buttons, progress bar, etc) should follow the 'Link' color
* Open the block settings sidebar and select a block that has some Sensei blocks, like the header or sidebar. Set the background, text and link color. Elements in the block should follow those colors.
* Save the template and open a lesson in the frontend. Colors should match the ones that were set.

### Notes
- Some blocks, like Notices, don't yet work with these colors

### Related tasks & follow-ups

- Add sensei prefix to css variables like `--primary-color`, so it won't conflict with anything. (I'll do a separate PR for this to avoid the extra noise in this one)
- Try to enable the Link color controls in classic themes too
- Go through all blocks and make sure styles work in the frontend and editor [#5734](https://github.com/Automattic/sensei/issues/5734)


### Screenshot / Video

https://user-images.githubusercontent.com/176949/191769418-2ace75e3-7307-4a8f-ad95-8160478aec03.mov
